core/src/main/java/net/opentsdb/query/BaseQueryNodeConfig.java

```diff
@@ -244,6 +251,35 @@ public HashCode buildHashCode() {
     return hc.hash();
   }
   
+  @Override
+  public List<QueryResultId> resultIds() {
+    return result_ids == null ? Collections.emptyList() : result_ids;
```

Comment:
As commented before, Let's just return `null`

Reply:
I tried the null route and it really is ugly. The static ref is a bit cleaner.

Reply:
Agreed.